doc/8.topics/fudi.html

```diff
@@ -0,0 +1,118 @@
+<!DOCTYPE html>
+<html lang="en-us">
+<head>
+<title>FUDI</title>
+<style>
+BODY { 
+	background:  #ffffff;;
+	color: 	     #000000;
+	font-family: Georgia, serif;
+	font-size:   12 pt; 
+	line-height: 1.6;
+}
+	
+#corpus {
+	width: 	    6.5in; 
+	margin-left: 0.8in;
+}
+
+H1, H2, H3, H4, H5, H6, ol, ul, mark, p, PRE { 
+	color: #3E4349; 
+}
+
+H1, H2 {
+	text-align:  center;
+	font-weight: normal;
+}
+
+PRE { 
+   font-size: 110%;
+   background-color:#f0f0f0;
+   text-align:  left;
+   border-radius: 8px;
+   padding: 8px;
+   font-family: 'Consolas', 'Menlo', 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', monospace;
+}
+
+/* standard link */
+
+a:link {
+    text-decoration: none;
+	color: #006699;
+}
+
+a:visited {
+    text-decoration: none;
+	color: #006699;
+}
+
+a:hover {
+
+	background-color:#eeeee4;
+}
+
+a:active {
+    text-decoration: none;
+	color: blue;
+}
+</style>
+<body>
+<div id=corpus>
+<h1 id="FUDI">FUDI</h1>
+
+<p>FUDI (<strong>F</strong>ast <strong>U</strong>niversal <strong>D</strong>igital <strong>I</strong>nterface) is a networking protocol used by the Pure Data patching language invented by Miller Puckette. It is a string based protocol in which messages are separated by semicolons. Messages are made up of tokens separated by whitespaces, and numerical tokens are represented as strings.</p>
+
+<h3 id="Format">Format</h3>
+
+<p><strong>FUDI</strong> is a packet oriented protocol.</p>
+
+<p>Each message consists of one or more <strong>atom</strong>s, separated by one or more <strong>whitespace</strong> characters, and it&rsquo;s terminated by a <strong>semicolon</strong> character.</p>
+
+<p>An <strong>atom</strong> is a sequence of one or more characters; whitespaces inside atoms can be escaped by the <strong>backslash</strong> (ascii 92) character (see Examples below).</p>
+
+<p>A <strong>whitespace</strong> is either a space (ascii 32), a tab (ascii 9) or a newline (ascii 10).</p>
+
+<p>A <strong>semicolon</strong> (ascii 59) is mandatory to terminate (and send) a message. A <strong>newline</strong> is just treated as whitespace and not needed for message termination.</p>
+
+<h3 id="Implementations">Implementations</h3>
+
+<p><strong>pdsend / pdreceive</strong></p>
+
+<p>Those command-line tools are distributed with the software Pure Data. They are meant to be used with their counterparts, the classes [netsend] / [netreceive] of Pd.</p>
+
+<p><strong>[netsend] / [netreceive]</strong></p>
+
+<p>Those classes can be used to transport Pd-messages over a TCP or UDP socket. Both are part of Pd-vanilla.</p>
+
+<p><strong>[netserver] / [netclient]</strong></p>
```

Comment:
should we drop [netserver] and [netclient] as reference implementations?
maxlib hasn't seen much love in the last years...

instead `[fudiparse]` and `[fudiformat]` should be mentioned here.

(if you want to keep `[netserver]`/`[netclient]`, you might want to separate it from built-in objects.

also, I like the article in PdPedia better than the WikiPedia, as it also expands on other languages (which is interesting if you want to interface your own little program with Pd) https://archive.ph/20100704034430/http://wiki.puredata.info/en/FUDI

Reply:
@umlaeute done with https://github.com/porres/pure-data/pull/22